external-providers/golang-dependency-provider/main.go

```diff
@@ -17,42 +19,47 @@ const (
 )
 
 // TODO implement this for real
-func (g *golangServiceClient) findGoMod() string {
-	var depPath string
-	if g.config.DependencyPath == "" {
-		depPath = "go.mod"
-	} else {
-		depPath = g.config.DependencyPath
-	}
-	f, err := filepath.Abs(filepath.Join(g.config.Location, depPath))
+func main() {
+	ll, err := GetDependenciesDAG()
 	if err != nil {
-		return ""
-	}
-	return f
-}
-
-func (g *golangServiceClient) GetDependencies() (map[uri.URI][]*provider.Dep, error) {
-	ll, err := g.GetDependenciesDAG()
-	if err != nil {
-		return nil, err
+		log.Fatal(err)
+		return
 	}
 	if len(ll) == 0 {
-		return nil, nil
+		log.Fatal(fmt.Errorf("no dependencies"))
+		return
 	}
 
-	m := map[uri.URI][]*provider.Dep{}
+	m := map[uri.URI][]provider.Dep{}
 	for u, d := range ll {
-		m[u] = provider.ConvertDagItemsToList(d)
+		m[u] = ConvertDagItemsToList(d)
```

Comment:
why not use the implementation in `provider`?

Reply:
i thought having a `*provider.Dep` would interfere with marshalling, I tried it out with the implementation in `provider` and it works, so ill change that.